internal/controller/pkg/revision/runtime_override_options.go

```diff
@@ -393,10 +411,19 @@ func ServiceWithSelectors(selectors map[string]string) ServiceOverride {
 	}
 }
 
-// ServiceWithAdditionalPorts adds additional ports to a Service.
+// ServiceWithAdditionalPorts adds additional ports to a Service if no port with the same name was found.
```

Comment:
is this "if no port with the same name was found" logic still needed for Services, since we no longer allow a ServiceTemplate spec? in which case would this logic still be needed now?

we do need the similar additional logic for Deployments, but maybe not for Services?

Reply:
i can also see a case being made for the behavior being consistent for both, to limit surprises for folks in this code next time...so just let me know what you think 😇